daemon/container_linux.go

```diff
@@ -513,13 +513,21 @@ func (container *Container) buildPortMapInfo(n libnetwork.Network, ep libnetwork
 		networkSettings.MacAddress = mac.(net.HardwareAddr).String()
 	}
 
+	networkSettings.Ports = nat.PortMap{}
+
+	if expData, ok := driverInfo[netlabel.ExposedPorts]; ok {
+		for _, tp := range expData.([]types.TransportPort) {
```

Comment:
It cannot happen, the code that packs driverInfo is vendored go code.
But ok, I can add the type assertion.